airflow/utils/cli.py

```diff
@@ -328,3 +344,32 @@ def _wrapper(*args, **kwargs):
                     logging.disable(logging.NOTSET)
 
     return cast(T, _wrapper)
+
+
+def get_config_with_source(include_default=False):
+    """
+    Return configuration along with source for each option.
+    """
+    config_dict = conf.as_dict(display_source=True)
+
+    with io.StringIO() as buf, redirect_stdout(buf):
+        for section, options in config_dict.items():
+            if not include_default:
+                options = {
+                    key: (value, source) for key, (value, source) in options.items() if source != "default"
+                }
+
+            # Print the section only when there are options after filtering
+            if options:
+                print(f"[{section}]")
+                for key, (value, source) in options.items():
+                    if not include_default and source == "default":
+                        continue
```

Comment:
```suggestion
```

We've already filtered these out, no?

Reply:
Thanks, will remove it. It was leftover code after comprehension.